github/commands/pull_request.py

```diff
@@ -121,12 +124,48 @@ def create_branch_for_pr(self, branch_name):
             self.pr["head"]["ref"]
         )
 
+        if not branch_name:
+            branch_name = self.pr["head"]["sha"]
+
         self.window.status_message("Creating local branch for PR...")
         self.git(
             "branch",
             branch_name,
             self.pr["head"]["sha"]
         )
+        sublime.set_timeout(lambda: self.set_upstream_for_branch(branch_name))
+
+    def set_upstream_for_branch(self, branch_name):
+        remotes = list(self.get_remotes().keys())
+        remote = self.pr["user"]["login"]
+        remote_branch = self.pr["head"]["ref"]
+
+        clone_url = self.pr["head"]["repo"]["clone_url"]
+        ssh_url = self.pr["head"]["repo"]["ssh_url"]
+
+        if remote not in remotes:
+            if not sublime.ok_cancel_dialog("Add remote '{}'?".format(remote)):
+                return
+
+            def on_select_url(index):
+                if index < 0:
+                    return
+                elif index == 0:
+                    url = clone_url
+                elif index == 1:
+                    url = ssh_url
+
+                self.git("remote", "add", remote, url)
+                self.git("fetch", remote, remote_branch)
+                self.git("branch", "-u", "{}/{}".format(remote, remote_branch), branch_name)
+
+            sublime.set_timeout(
+                lambda: self.window.show_quick_panel(
+                    [clone_url, ssh_url], on_select_url)
```

Comment:
I don't know what `clone_url` vs. `ssh_url` actually represent. I will check that on the functional tests later.

Reply:
That being said, Github recommends `https` urls these days because these are accessible and just work in most environments.

Reply:
`clone_url` is like "https://github.com/octocat/Hello-World.git" and `ssh_url` is like "git@github.com:octocat/Hello-World.git"